lightkurve/targetpixelfile.py

```diff
@@ -503,6 +534,87 @@ def estimate_centroids(self, aperture_mask='pipeline'):
             row_centr = np.nansum(yy * aperture_mask * self.flux, axis=(1, 2)) / total_flux
         return col_centr, row_centr
 
+    def _estimate_centroids_via_quadratic(self, aperture_mask):
+        """Estimate centroids by fitting a 2D quadratic to the brightest pixels;
+        this is a helper method for `estimate_centroids()`.
+
+        This method will fit a simple 2D second-order polynomial
+        $P(x, y) = a + bx + cy + dx^2 + exy + fy^2$
+        to the 3x3 patch of pixels centered on the brightest pixel within
+        the aperture mask.  This function approximates the core of the Point
+        Spread Function (PSF) using a bivariate quadratic function, and returns
+        the maximum (x, y) coordinate of the function using linear algebra.
+
+        For the motivation and the details around this technique, please refer
+        to Vakili, M., & Hogg, D. W. 2016, ArXiv, 1610.05873.
+
+        Parameters
+        ----------
+        aperture_mask : array-like, 'pipeline', or 'all'
+            A boolean array describing the aperture such that `True` means
+            that the pixel will be used.
+            If None or 'all' are passed, all pixels will be used.
+            If 'pipeline' is passed, the mask suggested by the official pipeline
+            will be returned.
+            If 'threshold' is passed, all pixels brighter than 3-sigma above
+            the median flux will be used.
+
+        Returns
+        -------
+        col_centr, row_centr : tuple
+            Arrays containing centroids for column and row at each cadence.
+        """
+        aperture_mask = self._parse_aperture_mask(aperture_mask)
+        col_centr, row_centr = [], []
+
+        # For each cadence (elow, we will fit the coefficients of the bivariate
+        # quadratic with the help of a design matrix (A) as defined by Eqn 20
+        # in Vakili & Hogg (arxiv:1610.05873). The design matrix contains a
+        # column of ones followed by pixel coordinates: x, y, x**2, xy, y**2.
+        A = np.array([[1, -1, -1, 1,  1, 1],
+                      [1,  0, -1, 0,  0, 1],
+                      [1,  1, -1, 1, -1, 1],
+                      [1, -1,  0, 1,  0, 0],
+                      [1,  0,  0, 0,  0, 0],
+                      [1,  1,  0, 1,  0, 0],
+                      [1, -1,  1, 1, -1, 1],
+                      [1,  0,  1, 0,  0, 1],
+                      [1,  1,  1, 1,  1, 1]])
+        # We also pre-compute $(A^t A)^-1 A^t$, cf. Eqn 21 in Vakili & Hogg.
+        At = A.transpose()
+        Aprime = np.linalg.inv(At.dot(A)).dot(At)  # np.linalg.inv(At @ A) @ At
```

Comment:
@barentsen I have a tiny objection here.  I would use ``matmul`` in place of ``dot``, see the second bullet in the
numpy docs https://docs.scipy.org/doc/numpy/reference/generated/numpy.dot.html